### PR TITLE
Use Cloud Posse's Utils provider to deep merge

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
* Uses Cloudposse Utils Provider to allow deepmerges

## why
* Need a way to merge with merging lists (exposed via `append_list` variable)

### CODE REVIEW

File: modules/yaml-config/variables.tf [hunk 1]
```diff
@@ -4,30 +4,12 @@ variable "map_config_local_base_path" {
...
-  description = "Base path to local YAML configuration files of list type"
-  default     = ""
-}
-
-variable "list_config_remote_base_path" {
-  type        = string
-  description = "Base path to remote YAML configuration files of list type"
-  default     = ""
-}
-
```
Comment (osterman) [inline modules/yaml-config/variables.tf:30]: We shouldn't be deprecating any variables

File: modules/yaml-config/variables.tf [hunk 2]
```diff
@@ -4,30 +4,12 @@ variable "map_config_local_base_path" {
   default     = ""
 }
 
-variable "map_config_remote_base_path" {
-  type        = string
-  description = "Base path to remote YAML configuration files of map type"
-  default     = ""
-}
-
```
Comment (osterman) [inline modules/yaml-config/variables.tf:12]: We shouldn't be deprecating any variables

File: modules/yaml-config/outputs.tf [hunk 3]
```diff
@@ -1,24 +1,4 @@
...
-
-output "remote_map_imports" {
-  value       = local.remote_map_imports
-  description = "Imports from remote map configs"
-}
-
-output "all_map_imports" {
-  value       = local.all_map_imports
-  description = "Combined imports from local and remote map configs"
-}
```
Comment (osterman) [inline modules/yaml-config/outputs.tf:24]: We shouldn't be deprecating any outputs

File: modules/deepmerge/variables.tf [hunk 4]
```diff
@@ -5,12 +5,24 @@ variable "maps" {
...
-    error_message = "The `maps` variable must be a list of maps and/or objects. Not all elements meet this requirement."
-  }
+  #  validation {
+  #    condition = ! can(index([
+  #      for mp in var.maps :
+  #      try(tolist(mp), toset(mp), tonumber(mp), tobool(mp), tostring(mp), null) == null
+  #      ]
+  #    , false))
+  #    error_message = "The `maps` variable must be a list of maps and/or objects. Not all elements meet this requirement."
+  #  }
```
Comment (nitrocode) [inline modules/deepmerge/variables.tf:15]: Why is this commented out ?

File: modules/deepmerge/variables.tf [hunk 5]
```diff
@@ -5,12 +5,24 @@ variable "maps" {
...
+}
+
+variable "append_list" {
```
Comment (korenyoni) [inline modules/deepmerge/variables.tf:18]: These should be suffixed with `_enabled` ideally (might need rewording). https://docs.cloudposse.com/reference/best-practices/terraform-best-practices/#use-feature-flags-to-enabledisable-functionality

File: modules/yaml-config/main.tf [hunk 6]
```diff
@@ -83,18 +92,26 @@ data "template_file" "remote_config" {
   vars     = var.parameters
 }
 
+module "deep_merge" {
+  source = "../deepmerge"
+  maps   = concat([{}], local.remote_map_configs_decoded, local.local_map_configs_decoded)
+  #  maps           = concat(local.remote_map_configs, local.local_map_configs)
```
Comment (nitrocode) [inline modules/yaml-config/main.tf:98]: Why comment this out instead of removing ?

File: modules/yaml-config/outputs.tf [hunk 7]
```diff
@@ -1,5 +1,6 @@
 output "map_configs" {
-  value       = module.all_map_configs.merged
+  #  value       = yamldecode(module.deep_merge.merged)
```
Comment (nitrocode) [inline modules/yaml-config/outputs.tf:2]: Why comment this out instead of removing ?

File: modules/yaml-config/main.tf (hunk 6 above)
Comment (Benbentwo) [reply modules/yaml-config/main.tf:98]: still testing things lol

**Review by osterman — APPROVED** [review]

**Review by nitrocode — DISMISSED** [review]

File: modules/yaml-config/main.tf [hunk 8]
```diff
@@ -5,7 +5,14 @@ locals {
   ] : []
 
   # Terraform maps from local YAML configuration templates
-  local_map_configs = flatten(
+  local_map_configs = flatten([
```
Comment (aknysh) [inline modules/yaml-config/main.tf:8]: `local_map_configs` looks like not used anywhere in the code

File: modules/yaml-config/main.tf [hunk 9]
```diff
@@ -21,7 +28,13 @@ locals {
   ] : []
 
   # Terraform maps from remote YAML configuration templates
-  remote_map_configs = flatten(
+  remote_map_configs = flatten([
```
Comment (aknysh) [inline modules/yaml-config/main.tf:31]: `remove_map_configs` looks like not used anywhere in the code

**Review by aknysh — CHANGES_REQUESTED** [review]
@Benbentwo please see comments

File: modules/yaml-config/main.tf (hunk 8 above)
Comment (Benbentwo) [reply modules/yaml-config/main.tf:8]: Thanks, repurposed this local with the `_decoded` because of output types and ended up not using this after testing

**Review by aknysh — APPROVED** [review]

**Pull request merged**